kwok/cloudprovider/cloudprovider.go

```diff
@@ -133,49 +132,37 @@ func (c CloudProvider) toNode(nodeClaim *v1beta1.NodeClaim) (*v1.Node, error) {
 	//nolint
 	newName = fmt.Sprintf("%s-%d", newName, rand.Uint32())
 
-	capacityType := v1beta1.CapacityTypeOnDemand
-	requirements := scheduling.NewNodeSelectorRequirementsWithMinValues(nodeClaim.
-		Spec.Requirements...)
-	if requirements.Get(v1beta1.CapacityTypeLabelKey).Has(v1beta1.CapacityTypeSpot) {
-		capacityType = v1beta1.CapacityTypeSpot
-	}
+	requirements := scheduling.NewNodeSelectorRequirementsWithMinValues(nodeClaim.Spec.Requirements...)
 	req, found := lo.Find(nodeClaim.Spec.Requirements, func(req v1beta1.NodeSelectorRequirementWithMinValues) bool {
 		return req.Key == v1.LabelInstanceTypeStable
 	})
 	if !found {
 		return nil, fmt.Errorf("instance type requirement not found")
 	}
 
-	minInstanceTypePrice := math.MaxFloat64
 	var instanceType *cloudprovider.InstanceType
+	var bestOffering *cloudprovider.Offering
 	// Loop through instance type values, as the node claim will only have the In operator.
 	for _, val := range req.Values {
-		var price float64
-		var ok bool
 		it, err := c.getInstanceType(val)
 		if err != nil {
 			return nil, fmt.Errorf("instance type %s not found", val)
 		}
 
-		// Since we're constructing the instance types we know that each instance type with OD offerings will have spot
-		// offerings, where spot will always be cheapest.
-		zone := randomChoice(KwokZones)
-		offering, ok := it.Offerings.Get(capacityType, zone)
-		if !ok {
-			return nil, fmt.Errorf("failed to find offering %s/%s/%s", capacityType, zone, val)
-		}
-		// All offerings of the same capacity type are the same price.
-		price = offering.Price
-		if price < minInstanceTypePrice {
-			minInstanceTypePrice = price
+		availableOfferings := it.Offerings.Available().Compatible(requirements)
+
+		offeringsByPrice := lo.GroupBy(availableOfferings, func(of cloudprovider.Offering) float64 { return of.Price })
```

Comment:
Do we need this logic to group all of the offerings by price like this? Can we consider just iterating through the available offerings that are compatible one-by-one and just doing the same logic -- e.g. finding the cheapest offering of the lot

Reply:
Previous behaviour picked a random offering if there are multiple offerings of the same price.  I'm using the grouping logic to maintain parity with previous behaviour (which I think is probably correct?)